StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1520UnitTests.cs

```diff
@@ -0,0 +1,485 @@
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will work properly for an if .. else statement.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestCodeFixProviderForElseStatementAsync()
+        {
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0)
+        {
+            Debug.Assert(true);
+        }
+        else if (i == 3)
+        {
+            Debug.Assert(false);
+        }
+        else
+            Debug.Assert(false);
+    }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0)
+        {
+            Debug.Assert(true);
+        }
+        else if (i == 3)
+        {
+            Debug.Assert(false);
+        }
+        else
+        {
+            Debug.Assert(false);
+        }
+    }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will properly handle alternate indentations.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact(Skip = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/660")]
+        public async Task TestCodeFixProviderWithAlternateIndentationAsync()
+        {
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+ public void Bar(int i)
+ {
+  if (i == 0)
+   Debug.Assert(true);
+  else
+  {
+   Debug.Assert(false);
+  }
+ }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+ public void Bar(int i)
+ {
+  if (i == 0)
+  {
+   Debug.Assert(true);
+  }
+  else
+  {
+   Debug.Assert(false);
+  }
+ }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix provider will work properly handle non-whitespace trivia.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestCodeFixProviderWithNonWhitespaceTriviaAsync()
+        {
+            var testCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+#pragma warning restore
+        if (i == 0)
+            Debug.Assert(true);
+        else
+        {
+            Debug.Assert(false);
+        }
+    }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+#pragma warning restore
+        if (i == 0)
+        {
+            Debug.Assert(true);
+        }
+        else
+        {
+            Debug.Assert(false);
+        }
+    }
+}";
```

Comment:
would be nice to have some inline comments before and after statements to verify that they are handled properly.